extensions/wikia/AbPerfTesting/js/FrontendDelay.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Introduce a delayed onDOMReady event to measure the impact of performance regressions
+ *
+  * @see FrontendDelay.class.php
+ */
+(function(w, log, weppy, $) {
+
+	function updateTransactionName(perf_test) {
+		// update Weppy config when transaction context is updated
+		// matches \Transaction::PARAM_AB_PERFORMANCE_TEST const
+		w.wgTransactionContext['perf_test'] = perf_test;
+
+		weppy.setOptions({
+			context: w.wgTransactionContext
+		});
+
+		// set UA custom dimension
+		// should match the PHP code in /extensions/wikia/AbPerfTesting/classes/Hooks.class.php
+		if (perf_test) {
+			w._gaq.push(
+				['set', 'dimension20', perf_test]
+			);
+		}
+	}
+
+	function info(msg) {
+		log(msg, log.levels.info, 'FrontEndDelay');
+	}
+
+	var delay = w.wgPerfTestFrontEndDelay,
+		perf_test = w.wgTransactionContext['perf_test'];
+
+	// FIXME: perform the bucketing logic here, should be moved to the backend
+	delay = false;
+
+	if (typeof w.beacon_id === 'string') {
+		switch(w.beacon_id.charAt(0)) {
+			case '0':
```

Comment:
how about something more like instead of the switch:
```
var DELAY_BUCKETS = {
  '0': 0, 
  '1': 100, 
  '2': 200, 
  '3': 300
};
var i = w.beacon_in.charAt(0); // crc?
if ( i in DELAY_BUCKETS ) {
  perf_test += '_' + i;
  delay = DELAY_BUCKETS[i];
}
```